client/my-sites/checkout/get-thank-you-page-url/index.ts

```diff
@@ -533,14 +533,23 @@ function getFallbackDestination( {
 		}
 	}
 
-	const pluginSlugs =
-		cart?.products
-			?.filter( ( product ) => product?.extra?.is_marketplace_product )
-			?.map( ( product ) => product?.extra?.plugin_slug ) || [];
+	const marketplaceProducts =
+		cart?.products?.filter( ( product ) => product?.extra?.is_marketplace_product ) || [];
 
-	if ( pluginSlugs.length > 0 ) {
+	const marketplacePluginSlugs = marketplaceProducts
+		?.filter( ( { extra } ) => extra.product_type === 'marketplace_plugin' )
+		.map( ( { extra } ) => extra.product_slug );
+
+	const marketplaceThemeSlugs = marketplaceProducts
+		?.filter( ( { extra } ) => extra.product_type === 'marketplace_theme' )
+		.map( ( { extra } ) => extra.product_slug );
```

Comment:
```suggestion
	const marketplacePluginSlugs = marketplaceProducts
		.filter( ( { extra } ) => extra.product_type === 'marketplace_plugin' )
		.map( ( { extra } ) => extra.product_slug );

	const marketplaceThemeSlugs = marketplaceProducts
		.filter( ( { extra } ) => extra.product_type === 'marketplace_theme' )
		.map( ( { extra } ) => extra.product_slug );
```

Nit: As `marketplaceProducts` will always be an array (maybe empty but still) I think you can remove the conditional check (`?`)

Reply:
Thanks for pointing this out, I was planning to remove this at one point but received a TS error, went back to fix it, then forgot to remove the optional chaining.